app/models/manageiq/providers/ibm_power_hmc/inventory/parser/infra_manager.rb

```diff
@@ -164,11 +194,11 @@ def parse_vm_guest_devices(lpar, hardware)
   end
 
   def parse_lpar_guest_devices(lpar, vm)
-    hardware = nil
+    hardware ||= persister.hardwares.lazy_find(:vm_or_template => vm)
```

Comment:
Since this is the first time you're setting `hardware` you can do a strict `=` assignment rather than a `||=`

Also if you have access to where the hardware is build it would be preferable to pass that in as an arg rather than doing a lazy_find since that will incur an extra query